### DISCUSSION TITLE
ARP lookup: Timeout

### DISCUSSION THREAD

**Discussion by pseudotronics** [cesanta/mongoose#2944] [Ask questions, get answers] [answered]
I have an issue where calling `mg_connect` will immediately result in an `ARP Timeout` being returned from `mg_connect_resolved`. I believe this is caused because the interfaces time is not updated when calling `mg_connect` and is only updated when `mg_tcpip_poll` is called.

So any delay between the last call of `mg_tcpip_poll ` and `mg_connect ` is considered part of the ARP timeout. 

In my specific instance the timeout is actually placed in the past which causes the ARP to immediately timeout.

```
3cff   3 mongoose.c:5939:mg_connect_res 9 ARP lookup...
3cff   4 mongoose.c:5077:settmout       9 2 -> 3c89
3d00   1 mongoose.c:1363:mg_error       9 -1 ARP timeout
```
I can, of course, fix the issue by calling `mg_mgr_poll` immediately before connecting.

I can turn this into an issue if needed. I just figured I would mention it here first.

**Comment by scaprile** [comment, marked as answer]
This has been reworked and tested last week
It works OK if you call Mongoose functions as expected.
Please see our [documentation](https://mongoose.ws/documentation/),  and follow the guidelines in our [tutorials](https://mongoose.ws/documentation/#tutorials-and-examples).
You are expected to call mg_mgr_poll() as frequently as possible, all our embedded calls are in a tight infinite loop, as you can see in all our examples and tutorials.
